// youtube-search-api/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.default = YTSearch;

var _axios = require('axios');

var _axios2 = _interopRequireDefault(_axios);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_URL = 'https://www.googleapis.com/youtube/v3/search';

function YTSearch(options, callback) {
	if (!options.key) throw new Error('API KEY is required but it is undefined!');

	var params = {
		part: 'snippet',
		key: options.key,
		q: options.term,
		maxResults: !options.max ? 5 : options.max,
		type: 'video',
		videoEmbeddable: 'true'
	};

	_axios2.default.get(BASE_URL, { params: params }).then(function (response) {
		if (callback) callback(response.data.items);
	}).catch(function (response) {
		for (var i = 0; i < response.data.error.errors.length; i++) {
			console.error('error:' + response.data.error.code + '\n\t\t\t\t  message:' + response.data.error.message + ' \n\t\t\t\t  reason:' + response.data.error.errors[i].reason);
		}
	});
}
